pandas/_libs/src/vendored/numpy/datetime/np_datetime.c

```diff
@@ -483,14 +483,27 @@ npy_datetime npy_datetimestruct_to_datetime(NPY_DATETIMEUNIT base,
   if (base == NPY_FR_ns) {
     int64_t nanoseconds;
 
-    // for near-minimum timestamps, scaling microseconds to nanoseconds
-    // overflows but adding nanoseconds puts the timestamp back in a valid range
+    // Minimum valid timestamp in nanoseconds (1677-09-21 00:12:43.145224193).
     const int64_t min_nanoseconds = NPY_MIN_INT64 + 1;
+
+    // For near-minimum timestamps (1677-09-21 00:12:43.145224193 through
+    // 1677-09-21 00:12:43.145224999), scaling microseconds to nanoseconds
+    // overflows (1677-09-21 00:12:43.145224 -> 1677-09-21 00:12:43.145224000),
+    // but adding nanoseconds can put the timestamp back in a valid range for
+    // nanosecond parts >= 193.
+
+    // (min_nanoseconds / 1000 - 1) * 1000 would overflow, so do not scale.
+    // This happens if microseconds corresponds to 1677-09-21 00:12:43.145224.
     if (microseconds == min_nanoseconds / 1000 - 1) {
-      // calculate final nanoseconds from minimum without scaling microseconds
+      // Instead, use minimum nanosecond timestamp as base and offset it with
```

Comment:
```suggestion
      // For values within one microsecond of INT64_MIN, use minimum nanosecond timestamp as base and offset it with
```

I think the comments are helpful but can be consolidated a bit. With a change like this alone I think can get rid of L488-496

Reply:
I went with your suggestion and removed other comments that were stating the obvious. Let me know if that works. Thanks for your quick reviews and suggestions!